docs/extensions/satrecsv.py

```diff
@@ -0,0 +1,240 @@
+                header = table.lines[0]
+                # Is this a statement?
+                if header[1].text.strip() == "Statement":
+                    for line in table.lines[1:]:
+                        number = line[0].text.strip()
+                        statement = line[1].text.strip()
+                        # guidance = line[2].strip()
+                        importance = line[3].text.strip()
+                        row = [title, number, statement, importance]
+                        rows.append(row)
+                        for i, cell in enumerate(row):
+                            column_widths[i] = max(column_widths[i], len(str(cell)))
+
+        wb = Workbook()
+        ws = wb.active
+        ws.append(header_row)
+        for cell in ws[1]:
+            cell.font = Font(bold=True)
+
+        for row in rows:
+            ws.append(row)
+
+        for c, cell in enumerate(ws[1]):
+            ws.column_dimensions[cell.column_letter].width = column_widths[c]
+
+        buffer = BytesIO()
+        wb.save(buffer)
+        self.output = buffer.getvalue()
+
+
+class SatreCsvTranslator(TextTranslator):
+    """
+    Extract SATRE evaluation statements from the docs and save as tabular file.
+
+    sphinx.writers.text.TextTranslator converts a document into a single plain text file
+    https://github.com/sphinx-doc/sphinx/blob/v7.2.5/sphinx/writers/text.py#L384-L1305
+
+    This class extends TextTranslator to ignore everything apart from the relevant SATRE
+    evaluation tables.
+    """
+
+    builder: SatreCsvBuilder
+
+    def __init__(self, document: nodes.document, builder: SatreCsvBuilder) -> None:
+        super().__init__(document, builder)
+        self.interested = []
+
+    # Disable default wrapping
+    def end_state(
+        self,
+        wrap: bool = True,
+        end: Sequence[str] | None = ("",),
+        first: str | None = None,
+    ) -> None:
+        super().end_state(False, end, first)
+
+    def visit_section(self, node: Element) -> None:
+        self.new_state(0)
+        self._title_char = self.sectionchars[self.sectionlevel]
+        self.sectionlevel += 1
+        if self.sectionlevel == 2:
+            self.interested.append({"table": [], "section": None})
+
+    def depart_section(self, node: Element) -> None:
+        self.interested[-1]["section"] = self.states[-1][0]
+        self.sectionlevel -= 1
+        self.end_state()
+
+    def visit_table(self, node: Element) -> None:
+        if hasattr(self, "table"):
+            msg = "Nested tables are not supported."
+            # TODO: Outputting a warning here causes the CI check to fail since we use
+            # SPHINXOPTS=-W to fail on errors. logging.skip_warningiserror doesn't work
+            # :-(
+            logger.info(msg)
+            # with logging.skip_warningiserror(True):
+            #     logger.warning(msg)
+            # Ideally we'd raise an error since we shouldn't have any nested tables but
```

Comment:
I've no idea why this nested table warning/error is triggered, there's no sign of one in our docs.